### ISSUE TITLE
error when running apply_data_changes()

### ISSUE THREAD

**Issue by abaezacastro** [comses/megadapt#34]
Error: Columns `antiguedad_dist`, `antiguedad_dren`, `falta_dren`, `Interventions_Ab`, `Interventions_D`, ... must have a unique name

**Comment by cpritcha** [comment]
How are running the model?

If I checkout 4b09ae49a598ee0dd0f08e118427baeb852986df (currently the latest commit), then build and install the `megadaptr` package (Cntl+Shift+B), then run the main scenario `source('~/code/src/r/scenarios/main.R', echo=TRUE)` I don't get any errors.

**Comment by abaezacastro** [comment]
Do you also get the error with the final commit 4b09ae4? 
I'm still getting the error.

**Comment by abaezacastro** [comment]
When creating next_year_changes, the variable "falta_dren" is recorded twice. Not sure why.
![image](https://user-images.githubusercontent.com/6106739/55283076-2f054480-530f-11e9-87d0-b0e6887dcc1f.png)

**Comment by cmbarton** [comment]
If you are using tidyverse, dplyr has its own bind command. Might want to use it for consistency.

Michael

Michael Barton

**Comment by cpritcha** [comment]
I'm not getting that error message. My working directory is clean and I've checked out that commit. Do you have any pending changes? Have you reinstalled `megadaptr`?

**Comment by cpritcha** [comment]
@cmbarton are you talking about [bind_cols](https://dplyr.tidyverse.org/reference/bind.html)? I want to join and replace existing columns in the data frame. The code I'm using right now to do so is

```r
apply_data_changes <- function(data, changes, join_columns) {
  change_colnames = setdiff(colnames(changes), names(join_columns))

  data %>%
    dplyr::select(-change_colnames) %>%
    dplyr::inner_join(
      changes,
      by = join_columns
    )
}
```

**Comment by cmbarton** [comment]
No. I was talking about dplyr::bind instead of cbind.

Michael


Michael Barton

**Comment by cpritcha** [comment]
Ok. That would be good to change too.

**Comment by cpritcha** [comment]
Fixed in 3aefafc26d77aa5fe7b8ab3429093469d927a4c2
